File: bank-service/src/main/java/rs/edu/raf/model/credit/Credit.java
package rs.edu.raf.model.credit;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Credit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private CreditRequest creditRequest;
    private Long contractDate;
    private Long loanMaturityDate;
    private BigDecimal installmentAmount;
    private BigDecimal remainingDebt;
    private Long nextInstallmentDate;

}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/credit/Credit.kt
package org.example.bankservicekotlin.model.credit

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
class Credit(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    @ManyToOne
    val creditRequest: CreditRequest,
    val contractDate: Long,
    val loanMaturityDate: Long,
    var installmentAmount: BigDecimal,
    var remainingDebt: BigDecimal,
    var nextInstallmentDate: Long
) {
}